### DISCUSSION TITLE
multiple nextcloud instances on k8s using aio helm chart

### DISCUSSION THREAD

**Discussion by melbruki** [nextcloud/all-in-one#3818] [Questions] [answered]
Do all Nextcloud Talk instances in a Kubernetes (k8s) environment require separate public IP addresses if they are configured on their own domains and namespaces? Can a single public IP be utilized for multiple instances, considering that using an ingress might not be feasible for this purpose? The goal is to minimize the use of public IPs. Additionally, in a typical setup, each Nextcloud AIO instance requires a separate Apache and Talk port. Does this requirement also apply in a Kubernetes environment?

In my current setup, each Nextcloud Talk instance is assigned a public IP address via a load balancer. I configure the firewall to allow traffic on port 3478 and set up DNS to point to a domain (e.g., talk.example.com), which I then specify in the Talk section of Nextcloud's administration settings. This configuration works well for a single instance. However, I'm considering running multiple Nextcloud Talk instances. My question is: Do I need to allocate a separate public IP address for each instance, or is there an alternative approach that would allow me to use a single public IP address for multiple instances? This inquiry is particularly relevant as I aim to optimize the usage of public IP addresses in my setup.

**Comment by szaimen** [comment, marked as answer]
Hi, you could have another NAT in front of kubernetes that forwards each service to each instance and only exposes one public ip-address.

Apart from that you might be able to use something like https://github.com/yrutschle/sslh that runs as kubernetes pod that maps the services internally to each instance. See https://github.com/nextcloud/all-in-one/discussions/3576
